src/software/thunderscope/robot_communication.py

```diff
@@ -291,6 +311,7 @@ def __enter__(self) -> "self":
 
         """
         # Create the multicast listeners
+        # TODO: log a better error message if the interface not specified
```

Comment:
are you planning on addressing this in this PR or making a new one. If it's the latter, then make a new issue